frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.unit.spec.tsx

```diff
@@ -362,6 +445,181 @@ describe("ClickActionsPopover", function () {
     });
+
+    describe("ObjectDetailsZoomDrill", () => {
+      it.each([
+        {
+          column: createOrdersTotalDatasetColumn(),
+          columnName: createOrdersTotalDatasetColumn().name,
+          cellValue: ORDERS_ROW_VALUES.TOTAL,
+          expectedAction: {
+            type: ZOOM_IN_ROW,
+            payload: { objectId: ORDERS_ROW_VALUES.ID },
+          },
+        },
+      ])(
+        "should apply drill on $columnName cell click",
+        async ({ column, cellValue, expectedAction }) => {
+          const { props } = await setup({
+            clicked: {
+              column,
+              value: cellValue,
+              data: ORDERS_COLUMNS.map(column => ({
+                col: ORDERS_COLUMNS_MAP[
+                  column.name as keyof typeof ORDERS_ROW_VALUES
+                ],
+                value:
+                  ORDERS_ROW_VALUES[
+                    column.name as keyof typeof ORDERS_ROW_VALUES
+                  ],
+              })),
+            },
+          });
+
+          const drill = screen.getByText("View details");
+          expect(drill).toBeInTheDocument();
+
+          userEvent.click(drill);
+
+          expect(props.dispatch).toHaveBeenCalledTimes(3);
+          expect(props.dispatch).toHaveBeenCalledWith(expectedAction);
+        },
+      );
+    });
+
+    describe("ObjectDetailsFkDrill", () => {
+      it.each([
+        {
+          column: createOrdersUserIdDatasetColumn(),
+          columnName: createOrdersUserIdDatasetColumn().name,
+          cellValue: ORDERS_ROW_VALUES.USER_ID,
+          expectedCard: {
+            dataset_query: {
+              database: SAMPLE_DB_ID,
+              query: {
+                filter: [
+                  "=",
+                  [
+                    "field",
+                    PEOPLE.ID,
+                    {
+                      "base-type": "type/BigInteger",
+                    },
+                  ],
+                  ORDERS_ROW_VALUES.USER_ID,
+                ],
+                "source-table": PEOPLE_ID,
+              },
+              type: "query",
+            },
+            display: "table",
+          },
+        },
+        {
+          column: createOrdersProductIdDatasetColumn(),
+          columnName: createOrdersProductIdDatasetColumn().name,
+          cellValue: ORDERS_ROW_VALUES.PRODUCT_ID,
+          expectedCard: {
+            dataset_query: {
+              database: SAMPLE_DB_ID,
+              query: {
+                filter: [
+                  "=",
+                  [
+                    "field",
+                    PRODUCTS.ID,
+                    {
+                      "base-type": "type/BigInteger",
+                    },
+                  ],
+                  ORDERS_ROW_VALUES.PRODUCT_ID,
+                ],
+                "source-table": PRODUCTS_ID,
+              },
+              type: "query",
+            },
+            display: "table",
+          },
+        },
+      ])(
```

Comment:
IMO it's really hard to read such long and nested test cases. Also, it makes it harder to tell the difference between test cases

Would it be possible to simplify this? Maybe make some helper functions, extract common field references to constants / functions?